File: storage/db-core/src/main/java/com/coffee/gu/menu/OptionGroupEntity.java
package com.coffee.gu.menu;

import com.coffee.gu.BaseEntity;
import jakarta.persistence.Entity;
import jakarta.persistence.Table;

@Table(name = "option_group")
@Entity
public class OptionGroupEntity extends BaseEntity {

    private String name;
    private Boolean isExclusive;
    private Boolean isRequired;

    protected OptionGroupEntity() {}

    public OptionGroupEntity(String name, Boolean isExclusive, Boolean isRequired) {
        this.name = name;
        this.isExclusive = isExclusive;
        this.isRequired = isRequired;
    }

    public OptionGroup toModel() {
        return new OptionGroup(
                id,
                name,
                isExclusive,
                isRequired
        );
    }

    public String getName() {
        return name;
    }

    public Boolean getExclusive() {
        return isExclusive;
    }

    public Boolean getRequired() {
        return isRequired;
    }
}

File: storage/db-core/src/main/kotlin/com/coffee/gu/menu/OptionGroupEntity.kt
package com.coffee.gu.menu

import com.coffee.gu.BaseEntity
import jakarta.persistence.Entity
import jakarta.persistence.GeneratedValue
import jakarta.persistence.GenerationType
import jakarta.persistence.Id
import jakarta.persistence.Table

@Table(name = "option_group")
@Entity
class OptionGroupEntity @JvmOverloads constructor(
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    var id: Long = 0,
    val name: String,
    val isExclusive: Boolean,
    val isRequired: Boolean,
) : BaseEntity() {

    fun toModel(): OptionGroup {
        return OptionGroup(
            id = id,
            name = name,
            isExclusive = isExclusive,
            isRequired = isRequired,
        )
    }
}
